libevmasm/Assembly.cpp

```diff
@@ -74,6 +76,209 @@ unsigned Assembly::codeSize(unsigned subTagSize) const
+			solRequire(
+				++current != end(_code) &&
+					createAssemblyItemFromJSON(*current, m_sourceList) == Instruction::JUMPDEST,
+				AssemblyImportException,
+				"JUMPDEST expected after tag."
+			);
+	}
+}
+
+AssemblyItem Assembly::createAssemblyItemFromJSON(Json::Value const& _json, std::vector<std::string> const& _sourceList)
+{
+	solRequire(_json.isObject(), AssemblyImportException, "Supplied JSON is not an object.");
+	static set<string> const validMembers{"name", "begin", "end", "source", "value", "modifierDepth", "jumpType"};
+	for (auto const& member: _json.getMemberNames())
+		solRequire(
+			validMembers.count(member),
+			AssemblyImportException,
+			"Unknown member '" + member + "'. Valid members are " +
+			solidity::util::joinHumanReadable(validMembers, ", ") + "."
+		);
+	solRequire(isOfType<string>(_json["name"]), AssemblyImportException, "Member 'name' not of type string.");
+	solRequire(isOfTypeIfExists<int>(_json, "begin"), AssemblyImportException, "Optional member 'begin' not of type int.");
+	solRequire(isOfTypeIfExists<int>(_json, "end"), AssemblyImportException, "Optional member 'end' not of type int.");
+	solRequire(isOfTypeIfExists<int>(_json, "source"), AssemblyImportException, "Optional member 'source' not of type int.");
+	solRequire(isOfTypeIfExists<string>(_json, "value"), AssemblyImportException, "Optional member 'value' not of type string.");
+	solRequire(
+		isOfTypeIfExists<int>(_json, "modifierDepth"),
+		AssemblyImportException,
+		"Optional member 'modifierDepth' not of type int."
+	);
+	solRequire(
+		isOfTypeIfExists<string>(_json, "jumpType"),
+		AssemblyImportException,
+		"Optional member 'jumpType' not of type string."
+	);
+
+	string name = getOrDefault<string>(_json["name"], "");
+	solRequire(!name.empty(), AssemblyImportException, "Member 'name' was empty.");
+
+	SourceLocation location;
+	location.start = get<int>(_json["begin"]);
+	location.end = get<int>(_json["end"]);
+	int srcIndex = getOrDefault<int>(_json["source"], -1);
+	size_t modifierDepth = static_cast<size_t>(getOrDefault<int>(_json["modifierDepth"], 0));
+	string value = getOrDefault<string>(_json["value"], "");
+	string jumpType = getOrDefault<string>(_json["jumpType"], "");
+
+	auto updateUsedTags = [&](u256 const& data)
+	{
+		m_usedTags = max(m_usedTags, static_cast<unsigned>(data) + 1);
+		return data;
+	};
+
+	auto storeImmutableHash = [&](string const& _immutableName) -> h256
+	{
+		h256 hash(util::keccak256(_immutableName));
+		solAssert(m_immutables.count(hash) == 0 || m_immutables[hash] == _immutableName);
+		m_immutables[hash] = _immutableName;
+		return hash;
+	};
+
+	auto storeLibraryHash = [&](string const& _libraryName) -> h256
+	{
+		h256 hash(util::keccak256(_libraryName));
+		solAssert(m_libraries.count(hash) == 0 || m_libraries[hash] == _libraryName);
+		m_libraries[hash] = _libraryName;
+		return hash;
+	};
+
+	auto valueMustBeDefined = [&](string const& _name, string const& _value)
```

Comment:
I'd use a naming convention that makes it clear that this is just a wrapper over `solRequire()`:

```suggestion
	auto requireValueDefinedForInstruction = [&](string const& _name, string const& _value)
```

Same for the other validation below.

Reply:
yeah that improves the readability a bit